src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -51,8 +50,18 @@
     "critical": DetectorPriorityLevel.HIGH,
 }
 
-LEGACY_ACTION_FIELDS = ["integration_id", "target_display", "target_identifier", "target_type"]
-ACTION_DATA_FIELDS = ["type", "sentry_app_id", "sentry_app_config"]
+TYPE_TO_PROVIDER = {
```

Comment:
nit: perhaps use `ActionService.EMAIL` etc instead of the raw ints, so we know what they are mapping to